jerry-core/ecma/operations/ecma-function-object.c

```diff
@@ -219,8 +367,36 @@ ecma_op_create_function_object (ecma_object_t *scope_p, /**< function's scope */
   return func_p;
 } /* ecma_op_create_function_object */
 
+/**
+ * Function object creation operation.
+ *
+ * See also: ECMA-262 v5, 13.2
+ *
+ * @return pointer to newly created Function object
+ */
+ecma_object_t *
+ecma_op_create_simple_function_object (ecma_object_t *scope_p, /**< function's scope */
+                                       const ecma_compiled_code_t *bytecode_data_p) /**< byte-code array */
+{
+  return ecma_op_create_function_object (scope_p, bytecode_data_p, ECMA_BUILTIN_ID_FUNCTION_PROTOTYPE);
+} /* ecma_op_create_simple_function_object */
+
 #if ENABLED (JERRY_ES2015)
 
+/**
+ * GeneratorFunction object creation operation.
+ *
+ * See also: ECMA-262 v5, 13.2
+ *
+ * @return pointer to newly created Function object
+ */
+ecma_object_t *
+ecma_op_create_generator_function_object (ecma_object_t *scope_p, /**< function's scope */
+                                          const ecma_compiled_code_t *bytecode_data_p) /**< byte-code array */
+{
+  return ecma_op_create_function_object (scope_p, bytecode_data_p, ECMA_BUILTIN_ID_GENERATOR);
+} /* ecma_op_create_generator_function_object */
+
```

Comment:
Do we need these methods at all? Can't we just use the `..._create_function_object` directly?

Reply:
Yes they can be called directly, but I thinks these helper functions are inlined by the compiler. I don't insist to this methods, but the used API has not changed only has been renamed (that was the institution).